virtualmachine/vsphere/util.go

```diff
@@ -586,9 +595,21 @@ var cloneFromTemplate = func(vm *VM, dcMo *mo.Datacenter, usableDatastores []str
 	// to delete all the network cards and create VirtualDevice specs.
 	// For now only configure the datastore and the host.
 	relocateSpec := types.VirtualMachineRelocateSpec{
-		Pool:      &l.ResourcePool,
-		Host:      &l.Host,
-		Datastore: &dsMor,
+		Pool: &l.ResourcePool,
+	}
+
+	crMo, err := findClusterComputeResource(vm, dcMo, vm.Destination.DestinationName)
+	if err != nil {
+		return err
+	}
+
+	drsEnabled := *crMo.Configuration.DrsConfig.Enabled
+
+	if vm.Destination.HostSystem != "" || !drsEnabled {
+		relocateSpec.Host = &l.Host
+	}
+	if dsMo != nil {
```

Comment:
do we need to error out in case of else here?

Reply:
The datastore field is optional, dsMo will be nil if datastore is not passed.